src/platform.ts

```diff
@@ -87,10 +87,7 @@ export class SwitchmateSwitchPlatform implements DynamicPlatformPlugin {
 			if (!that.discoveredAll) {
```

Comment:
This line would prevent the plugin from re-scanning forever when all configured devices are found. As a future improvement idea, maybe add a maximum retry count?